rotkehlchen/accounting/importer/json.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+from pathlib import Path
+from typing import Tuple
+
+from marshmallow import EXCLUDE, ValidationError
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.api.v1.schemas import (
+    HistoryBaseEntrySchema,
+    IgnoredActionsModifySchema,
+    ModifiableSettingsSchema,
+)
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JSONImporter:
+    def __init__(self, db: DBHandler) -> None:
+        self.db = db
+
+    def import_history_debug(self, filepath: Path) -> Tuple[bool, str]:
+        """Imports the user events, settings & ignored actions identifiers for debugging."""
+        try:
+            with open(filepath) as f:
+                debug_data = json.load(f)
+        except (PermissionError, json.JSONDecodeError) as e:
+            return False, str(e)
+
+        has_required_keys = (
+            'events' in debug_data or
+            'settings' in debug_data or
+            'ignored_events_ids' in debug_data
+        )
+        if has_required_keys is False:
+            return False, 'import history data contains invalid keys'
+        has_required_types = (
+            isinstance(debug_data['events'], list) or
+            isinstance(debug_data['settings'], dict) or
+            isinstance(debug_data['ignored_events_ids'], dict)
+        )
+        if has_required_types is False:
+            return False, 'import history data contains invalid data types'
+
+        # add history events.
+        log.debug('Trying to add history events')
+        history_db = DBHistoryEvents(self.db)
+        try:
+            for event in debug_data['events']:
+                event = HistoryBaseEntrySchema(identifier_required=False).load(event)
+                history_db.add_history_event(event['event'])
+        except ValidationError as e:
+            log.error(f'Error while adding history events due to: {str(e)}')
+            return False, 'invalid history event data'
+        except sqlcipher.DatabaseError as e:  # pylint: disable=no-member
+            history_db.db.conn.rollback()
+            error_msg = f'Error while adding history events due to: {str(e)}'
+            log.error(error_msg)
+            return False, error_msg
+        log.debug('Successfully added history events')
+
```

Comment:
Just noticed. I would say that there should be a commit to db in case of successful import, shouldn't it?

Reply:
The methods called call `self.update_last_write()` of the DBHandler.

Reply:
Okay. Then this probably won't work. Won't work since you cal `rollback()` after a commit with bad changes was made. I'm not sure on 100% but probably this `rollback()` will roll back to this freshly created commit and thus won't actually do anything.

Reply:
Errors/exceptions are raised on` .execute()` not on `commit()`. Commit simply makes the transaction visible to other db connections.

Reply:
Ah yes yes, of course, I said a stupid thing. But this is about my first comment, what do you think about the second one?

Reply:
> For example, `events` are successfully imported and a commit was made, but there was something wrong with settings or ignored events. In this case I guess you should want to abort the entire import, right? And committing between the steps doesn't make sense then, does it?

You do have a point but then that'd be quite impossible as each of those steps is in different transactions.
I mean they all call `self.update_last_write()` at the end of their operations.

One thing to note though is that this isn't used by users. It's for dev only.

Reply:
Yeah, I got you, I see the problem. I would say that one thing that is needed is to clear the events database right before importing a new json file. You don't have it yet, right? I think that it would makes sense since when we try to debug pnl we want to fully replicate a user's situation (and it also should help to avoid the problem with partially imported data). What do you think about it?

Reply:
I'm not so sure about this. I assume that to replicate a user pnl error, a new account will be created. What's your take?

Reply:
Mm, makes sense. I thought it is like 2 - 5 lines of code, or no?. In my opinion clearing the db is a nice-to-have thing because it most certainly won't break anything but might help once. And if it's easy to do, why not? (if difficult, sure, let's keep it as is)

Reply:
It doesn't cost anything to add though. Will add it

Reply:
Awesome!!  :partying_face: :partying_face: :partying_face: